src/material-experimental/mdc-form-field/form-field.ts

```diff
@@ -501,4 +551,52 @@ export class MatFormField implements AfterViewInit, OnDestroy, AfterContentCheck
       this._control.setDescribedByIds(ids);
     }
   }
+
+  /**
+   * Updates the horizontal offset of the label in the outline appearance. In the outline
+   * appearance, the notched-outline and label are not relative to the infix container because
+   * the outline intends to surround prefixes, suffixes and the infix. This means that the
+   * floating label by default overlaps prefixes in the docked state. To avoid this, we need to
+   * horizontally offset the label by the width of the prefix container. The MDC text-field does
+   * not need to do this because they use a fixed width for prefixes. Hence, they can simply
+   * incorporate the horizontal offset into their default text-field styles.
+   */
+  private _updateOutlineLabelOffset() {
+    if (!this._hasOutline() || !this._prefixContainer || !this._floatingLabel) {
+      return;
+    }
+    // If the form-field is not attached to the DOM yet (e.g. in a tab), we defer
+    // the label offset update until the zone stabilizes.
+    if (!this._isAttachedToDOM()) {
+      this._needsOutlineLabelOffsetUpdateOnStable = true;
+      return;
+    }
+
+    const floatingLabel = this._floatingLabel.element;
+    const prefixContainer = this._prefixContainer.nativeElement as HTMLElement;
+    // If the directionality is RTL, the x-axis transform needs to be inverted. This
+    // is because `transformX` does not change based on the page directionality.
+    const labelHorizontalOffset =
+      (this._dir.value === 'rtl' ? -1 : 1) * prefixContainer.getBoundingClientRect().width;
```

Comment:
We could also account for cases where the prefix containers start of invisible here, but I think that this is out-of-scope and very unlikely (it will also affect performance). 

I saw that something like this has been done in the non-MDC based form-field for the outline gap. Though IMO we should not complicate these things, and rather provide/expose  methods, so that people can refresh if needed.

Reply:
Am I reading this incorrectly, or is this calling `getBoundingClientRect` on every zone stable?

Reply:
`getBoundingClientRect` will be called only if the projected prefixes change, or if the appearance is set to `outline` from `fill`. The goal is to call this as less as possible.

Reply:
I see, you're guarding it on the boolean. It seems like it would be more idiomatic rxjs to combine the `_prefixChildren.changes` stream with the zone stable stream somehow (though I don't actually know the right operators off the top of my head)

Reply:
I'm unsure if it's achievable with RxJS in a less complicated way. Combining both observables would complicate things since we also want to update if the appearance input changes.

We need logic like: Whenever prefix changes, or the appearance input changes, and the Zone becomes stable, update the offset. And important would be to only update once if both prefix and input change. FWIW: The same pattern is used in the non MDC-based form-field.